image-download

```diff
@@ -85,6 +85,17 @@ def find(name, stores, latest, quiet):
     for store in stores:
         url = urllib.parse.urlparse(urllib.parse.urljoin(store, name))
 
+        # First, try the easy case: for OpenShift proxied stores which send
+        # their own SSL cert initially host name has to match that
+        args = ['--cacert', CA_PEM, url.geturl()]
+        output = check_curl_args(args)
+        if output:
+            show_status(quiet, 'present', store)
+            found.append((args, output, store))
+            continue
+
+        # Otherwise, the server is expecting the hostname "cockpit-tests".
```

Comment:
The server couldn't care less, it's just that our [certificate](https://github.com/cockpit-project/cockpituous/blob/master/tasks/credentials/openssl.cnf#L54) has `cockpit-tests` as a valid subjectAltName for image servers which don't have a predictable DNS name (such as we used to have with our AWS instance until a few weeks ago). I'm not actually sure if we still need this fallback case, now that all our image stores have a predictable name.

However, we can clean this up with moving to S3, when we can hopefully just drop this entire hackery. :grin: